diff -x '*.orig' -burN dmg2img-1.6.2.orig/Makefile dmg2img-1.6.2/Makefile
--- dmg2img-1.6.2.orig/Makefile	2012-01-19 18:03:25 +0000
+++ dmg2img-1.6.2/Makefile	2012-01-19 18:12:20 +0000
@@ -1,11 +1,11 @@
 #CC = gcc
-CFLAGS = -O2 -Wall
-BIN_DIR = ${DESTDIR}/usr/bin
+CFLAGS := ${CFLAGS} -O2 -Wall
+BIN_DIR = ${DESTDIR}/bin
 
 all: dmg2img vfdecrypt 
 
 dmg2img: dmg2img.c dmg2img.h mntcmd.h gpt.h dmg2img.o base64.o adc.o 
-	$(CC) -s -o dmg2img dmg2img.o base64.o adc.o -L. -lz -lbz2
+	$(CC) $(LDFLAGS) -s -o dmg2img dmg2img.o base64.o adc.o -L. -lz -lbz2
 
 dmg2img.o: dmg2img.c dmg2img.h
 	$(CC) $(CFLAGS) -c dmg2img.c
@@ -17,11 +17,11 @@
 	$(CC) $(CFLAGS) -c adc.c
 
 vfdecrypt: vfdecrypt.c
-	$(CC) $(CFLAGS) -s -o vfdecrypt vfdecrypt.c -lcrypto
+	$(CC) ${LDFLAGS} $(CFLAGS) -s -o vfdecrypt vfdecrypt.c -lcrypto
 
 install: dmg2img vfdecrypt
 	mkdir -p ${BIN_DIR}
-	install -c -s -m 755 -o root -g root dmg2img vfdecrypt $(BIN_DIR)
+	install -c -s -m 755 dmg2img${exeext} vfdecrypt${exeext} $(BIN_DIR)
 
 clean:
 	rm -f dmg2img vfdecrypt *~ *.o core
diff -x '*.orig' -burN dmg2img-1.6.2.orig/dmg2img.h dmg2img-1.6.2/dmg2img.h
--- dmg2img-1.6.2.orig/dmg2img.h	2012-01-19 18:03:25 +0000
+++ dmg2img-1.6.2/dmg2img.h	2012-01-19 18:03:25 +0000
@@ -20,6 +20,10 @@
 #include "adc.h"
 #include <unistd.h>
 
+#if defined(__MINGW32__)
+    #define fseeko _lseeki64
+#endif
+
 #define BT_ADC   0x80000004
 #define BT_ZLIB  0x80000005
 #define BT_BZLIB 0x80000006
diff -x '*.orig' -burN dmg2img-1.6.2.orig/vfdecrypt.c dmg2img-1.6.2/vfdecrypt.c
--- dmg2img-1.6.2.orig/vfdecrypt.c	2012-01-19 18:03:25 +0000
+++ dmg2img-1.6.2/vfdecrypt.c	2012-01-19 18:03:25 +0000
@@ -35,7 +35,15 @@
 #include <string.h>
 #include <unistd.h>
 #include <inttypes.h>
+#if defined(__MINGW32__)
+#define htonl(_x) ((((uint32_t)(_x) & 0xff000000) >> 24) | \
+                   (((uint32_t)(_x) & 0x00ff0000) >> 8)  | \
+                   (((uint32_t)(_x) & 0x0000ff00) << 8)  | \
+                   (((uint32_t)(_x) & 0x000000ff) << 24))
+#define ntohl(_x) htonl(_x)
+#else
 #include <arpa/inet.h>
+#endif
 #include <openssl/sha.h>
 #include <openssl/aes.h>
 #include <openssl/hmac.h>
