diff --git a/src/lns/Makefile b/src/lns/Makefile
index acee948..87025d9 100644
--- a/src/lns/Makefile
+++ b/src/lns/Makefile
@@ -1,7 +1,7 @@
 ####### Compiler, tools and options
 CXX           = g++
 DEFINES       =
-CXXFLAGS      = -pipe -Wall -W -D_REENTRANT
+CXXFLAGS     += -pipe -Wall -W -D_REENTRANT
 LINK          = g++
 LFLAGS        =
 TARGET        = ../../lns
diff --git a/src/lns/lns.cpp b/src/lns/lns.cpp
index fb95878..593b353 100644
--- a/src/lns/lns.cpp
+++ b/src/lns/lns.cpp
@@ -10,6 +10,9 @@
 
 #if defined(WIN64) || defined(_WIN64) || defined(__WIN64__) || defined(WIN32) || defined(_WIN32) || defined(__WIN32__) || defined(__NT__)
   #define ON_WINDOWS
+  #if defined(__MINGW32__)
+    #define _WIN32_WINNT 0x0600
+  #endif
   #include "dirent.h"  // includes windows.h for us
   #include <direct.h>
   #define DIRSEP '\\'
@@ -18,6 +21,8 @@
   { return CreateHardLinkA(path2, path1, NULL) ? 0 : -1; }
   int symlink(const char *path1, const char *path2)
   { return CreateSymbolicLinkA(path2, path1, 0) ? 0 : -1; }
+  int symlink_dir(const char *path1, const char *path2)
+  { return CreateSymbolicLinkA(path2, path1, 1) ? 0 : -1; }
   int mkdir(const char *path, int)
   { return mkdir(path); }
 #else
@@ -260,6 +265,10 @@ int main(int argc, char *argv[])
             exit(2);
         }
     }
+
+    if (!useHardlinks) {
+        if (!symlink_dir(source_path, destination_path)) return 0;
+    }
 #endif
 
     if (verboseness > 3)
