--- ./misc/libtool.c.orig	2009-06-11 12:30:58.000000000 -0500
+++ ./misc/libtool.c	2009-06-11 12:31:02.000000000 -0500
@@ -1035,6 +1037,11 @@
 	else{
 	    next_root = getenv("NEXT_ROOT");
 	}
+	if(next_root == NULL) {
+#ifdef CROSS_SYSROOT
+          next_root = CROSS_SYSROOT;
+#endif
+	}
 	if(next_root != NULL){
 	    for(i = 0; standard_dirs[i] != NULL; i++){
 		p = allocate(strlen(next_root) +
--- ./ld/ld.c.orig	2009-06-11 12:31:01.000000000 -0500
+++ ./ld/ld.c	2009-06-11 12:31:02.000000000 -0500
@@ -1916,6 +1918,11 @@
 	else{
 	    next_root = p;
 	}
+       if(next_root == NULL) {
+#ifdef CROSS_SYSROOT
+         next_root = CROSS_SYSROOT;
+#endif
+       }
 	if(next_root != NULL){
 	    for(i = 0; standard_dirs[i] != NULL; i++){
 		p = allocate(strlen(next_root) +
