--- ./read.c.orig	2012-01-09 00:30:54.703260218 +0000
+++ ./read.c	2012-01-09 00:31:57.290473379 +0000
@@ -37,6 +37,7 @@
 #include <stdlib.h>
 #include <sys/types.h>
 #include <sys/stat.h>
+#include <inttypes.h>
 #include "stuff/rnd.h"
 #include "stuff/arch.h"
 #include "stuff/best_arch.h"
@@ -2555,7 +2556,7 @@
 
 	*p = 0;
 	if(strlen(segname) > sizeof(s.segname)){
-	    as_bad("segment-name: %s too long (maximum %ld characters)",
+	    as_bad("segment-name: %s too long (maximum %" PRIdPTR " characters)",
 		    segname, sizeof(s.segname));
 	    ignore_rest_of_line();
 	    *p = c;
@@ -2564,7 +2565,7 @@
 
 	*q = 0;
 	if(strlen(sectname) > sizeof(s.sectname)){
-	    as_bad("section-name: %s too long (maximum %ld characters)",
+	    as_bad("section-name: %s too long (maximum %" PRIdPTR " characters)",
 		    sectname, sizeof(s.sectname));
 	    ignore_rest_of_line();
 	    return;
@@ -2729,7 +2730,7 @@
 
 	    *p = 0;
 	    if(strlen(segname) > sizeof(s.segname)){
-		as_bad("segment-name: %s too long (maximum %ld characters)",
+		as_bad("segment-name: %s too long (maximum %" PRIdPTR " characters)",
 			segname, sizeof(s.segname));
 		ignore_rest_of_line();
 		*p = c;
@@ -2738,7 +2739,7 @@
 
 	    *q = 0;
 	    if(strlen(sectname) > sizeof(s.sectname)){
-		as_bad("section-name: %s too long (maximum %ld characters)",
+		as_bad("section-name: %s too long (maximum %" PRIdPTR " characters)",
 			sectname, sizeof(s.sectname));
 		ignore_rest_of_line();
 		*p = c;
