--- execute.c.orig	2012-03-22 12:40:52 +0000
+++ execute.c	2012-03-22 12:53:25 +0000
@@ -27,7 +27,11 @@
 #include <mach/mach.h> /* first to get rid of pre-comp warning */
 #include "stdio.h"
 #include <signal.h>
+#if defined(__MINGW32__)
+#include <process.h>
+#else
 #include <sys/wait.h>
+#endif
 #include <sys/file.h>
 #include "stuff/errors.h"
 #include "stuff/allocate.h"
@@ -62,7 +66,10 @@
 		    fprintf(stderr, "%s ", *p++);
 	    fprintf(stderr, "\n");
 	}
-
+#if defined __MINGW32__
+    termsig = _spawnvp(_P_WAIT, name, (const char * const*)argv);
+    return(termsig==0);
+#else
 	forkpid = fork();
 	if(forkpid == -1)
 	    system_fatal("can't fork a new process to execute: %s", name);
@@ -91,6 +98,7 @@
 #endif
 		termsig == 0);
 	}
+#endif
 }
 
 /*
